File: python/1561.py
# Link: https://leetcode.com/problems/maximum-number-of-coins-you-can-get/description/
# Runtime: 507 ms
# Memory: 28.68 MB

class Solution:
    def maxCoins(self, piles: List[int]) -> int:
        sorted_piles = piles[:]
        sorted_piles.sort()

        return sum(sorted_piles[len(sorted_piles) // 3::2])

File: c++/1561.cpp
// Link: https://leetcode.com/problems/maximum-number-of-coins-you-can-get/description/
// Runtime: 98 ms
// Memory: 56.28 MB

class Solution {
public:
  int maxCoins(const std::vector<int> &piles) {
    std::vector<int> sortedPiles = piles;
    std::sort(sortedPiles.begin(), sortedPiles.end());

    int maxCoins = 0;
    for (int i = static_cast<int>(sortedPiles.size()) / 3;
         i < static_cast<int>(sortedPiles.size()); i += 2)
      maxCoins += sortedPiles[i];

    return maxCoins;
  }
};
